### PR TITLE
Fixes to support numpy values

### LINKED ISSUE #27: deep_type throws error on list of numpy ints
The following code causes deep_type to throw a ValueError:

```
import pytypes
import numpy as np
pytypes.deep_type([np.int64(1), np.int64(1)])
```

### PR DESCRIPTION
**Pull request by mitar**
Fixes #27.

### CODE REVIEW

File: pytypes/type_util.py [hunk 1]
```diff
@@ -282,6 +282,15 @@ def deep_type(obj, depth = None, max_sample = None):
     return _deep_type(obj, [], 0, depth, max_sample)
 
 
+def _is_in(obj, lst):
+    """Checks if obj is in lst using referential equality.
+    """
+    for el in lst:
+        if el is obj:
+            return True
+    return False
```
Comment (eric-wieser) [inline pytypes/type_util.py]: `return any(el is obj for el in lst)`

Comment (mitar) [reply pytypes/type_util.py]: Great oneliner!

**Comment by Stewori** [comment]
Just (minor) remarks:
 * ``_is_in`` should better be hosted in pytypes/util as it is not specific to type checking or typing at all (I just want to avoid to blow up ``type_util`` too much over time)

* I originally had this in mind:
```
def _obj_in_list(obj, lst):
	"""Simply using ``in`` is sometimes not reliable, e.g. for ``numpy.int64``,
	see https://github.com/Stewori/pytypes/issues/27.
	"""
	try:
		return obj in lst
	except ValueError:
		pass
	# try it for NumPy behavior
	try:
		return any(obj == lst)
	except ValueError:
		pass
	# try it using identity checks
	return any(el is obj for el in lst)
```

because I think both ``in`` and NumPy-level ``==`` iterate on C-level while ``for`` iterates on Python level. However I admit the short version with ``is`` is clearer and cleaner, especially for that purpose, so I'm fine with it.

**Comment by mitar** [comment]
I think also referential equality is much faster to check.

I will move things around as you suggested.

**Comment by mitar** [comment]
Updated.

**Pull request merged**